core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -187,6 +189,26 @@ private SdkClientConfiguration finalizeAwsConfiguration(SdkClientConfiguration c
                             .build();
     }
 
+    /**
+     * Apply the client override configuration to the provided configuration.
+     */
+    protected SdkClientConfiguration setOverrides(SdkClientConfiguration configuration) {
+        if (overrideConfig == null) {
+            return configuration;
+        }
+        SdkClientConfiguration.Builder builder = configuration.toBuilder()
+                                                              .putAll(overrideConfig);
+        overrideConfig.retryStrategy().ifPresent(retryStrategy -> builder.option(RETRY_STRATEGY, retryStrategy));
+        overrideConfig.retryMode().ifPresent(retryMode -> builder.option(RETRY_STRATEGY,
+                                                                         AwsRetryStrategy.forRetryMode(retryMode)));
+        overrideConfig.retryStrategyConfigurator().ifPresent(configurator -> {
+            RetryStrategy.Builder<?, ?> defaultBuilder = AwsRetryStrategy.defaultRetryStrategy().toBuilder();
+            configurator.accept(defaultBuilder);
+            builder.option(RETRY_STRATEGY, defaultBuilder.build());
+        });
```

Comment:
Should we not nullify the configured options here as well as in the parent method implementation? If not, what condition differs?

Reply:
Yes, we should, let me add a change for that.